rpm/dnf/libdnf_wrapper.cpp

```diff
@@ -0,0 +1,236 @@
+// +build dnf
+
+#include "libdnf_wrapper.h"
+#include <exception>
+#include <string>
+#include <string.h>
+
+const char* newCString(std::string s) {
+    char* msg = new char[s.size()+1];
+    strcpy(msg, s.c_str());
+    return msg;
+}
+
+const char* getErrorMessage(GError* gerr) {
+    if (gerr == nullptr)
+        return newCString("unknown error");
+
+    const char* msg = strdup(gerr->message);
+    g_error_free(gerr);
+    return msg;
+}
+
+const char* getErrorMessage(std::string prefix, GError* gerr) {
+    if (gerr == nullptr)
+        return newCString(prefix + "unknown error");
+
+    size_t prefix_size = prefix.size();
+    char* msg = new char[prefix_size + strlen(gerr->message) + 1];
+
+    strcpy(msg, prefix.c_str());
+    strcpy(msg + prefix_size, gerr->message);
+
```

Comment:
you've done the arithmetic, so it _should_ be OK, but I would suggest
strncpy(), and then strncat() to ensure you don't walk off your buffer.